src/components/map/map.js

```diff
@@ -86,60 +91,106 @@ export default function Map({
     })),
   };
 
+  const opoGeoJson = {
+    ...dsaGeoData.childGeoJson,
+    features: (opo
+      ? [findOpoFeature(dsaGeoData, opo)]
+      : dsaGeoData.childGeoJson.features
+    ).map(f => ({
+      ...f,
+      properties: {
+        ...f.properties,
+        tier: opoDataMap[f.properties.opo].tier,
+      },
+    })),
+  };
+
   // Map bounding box: individual state or continential US (first 48)
-  const [minX, minY, maxX, maxY] = bbox({
-    ...stateGeoJson,
-    features: stateGeoJson.features,
-  });
+  const [minX, minY, maxX, maxY] = bbox(
+    opo
+      ? {
+          ...opoGeoJson,
+          features: opoGeoJson.features,
+        }
+      : {
+          ...stateGeoJson,
+          features: stateGeoJson.features,
+        }
+  );
 
   return (
     <Row className={styles.map}>
       <div style={dimensions}>
         {legend && <Legend />}
-        {state ? null : <hr />}
+        {state || opo ? null : <hr />}
         {
           // Hack: [`window` dependency for Leaflet](https://www.gatsbyjs.com/docs/debugging-html-builds/#fixing-third-party-modules)
           typeof window !== "undefined" && (
             <MapContainer
-            key={state + "container"}
+              key={
+                state
+                  ? state + "container"
+                  : opo
+                  ? opo + "container"
+                  : "state container"
+              }
               bounds={[
                 [minY, minX],
                 [maxY, maxX],
               ]}
               scrollWheelZoom={false}
-              style={dimensions}
+              style={Object.assign(dimensions, {backgroundColor: "#fff"})}
               zoomControl={false}
               dragging={windowWidth > 800}
               className={styles.mapContainer}
             >
               {zoomControl && <ZoomControl position="bottomright" />}
               <GeoJSON
-                key={state + "opos"}
-                data={(state
-                  ? dsaGeoData?.childGeoJson?.features.filter(
-                      f =>
-                        opoDataMap[f.properties.opo].statesWithRegions[
-                          state
-                        ] !== undefined
-                    )
-                  : dsaGeoData?.childGeoJson?.features
-                ).map(f => ({
-                  ...f,
-                  properties: {
-                    ...f.properties,
-                    tier: opoDataMap[f.properties.opo].tier,
-                  },
-                }))}
+                key={state ? state + "opos" : opo ? opo + "opos" : "state opos"}
+                data={
```

Comment:
I find this ternary + object expansion hard to make sense of; would like comment that explains what is being composed + why (i.e. what logic is used in conditionals and what's the end result)